webpack.common.js

```diff
@@ -77,163 +78,177 @@ console.log('npm run watch  --themes_dir=themes/mytheme/client --css_file=myfile
-        JS_FILE,
-        CSS_FILE
-      ]
-    },
-    output: {
-      filename: '[name].js',
-      path: path.resolve(
-        DIST_DIR
-      )
-      // crossOriginLoading: 'anonymous'
-    },
-    module: {
-      rules: [
-        {
-          // javascripts
-          test: /\.jsx?$/,
-          // exclude: /node_modules/,
-          use: {
-            loader: 'babel-loader',
-            options: {
-              presets: [
-                '@babel/preset-env',
-                '@babel/react',
-                {
-                  plugins: [
-                    '@babel/plugin-proposal-class-properties'
-                  ]
-                }
-              ],
+const myConfig = merge({
+  // webpack cache system
+  cache: {
+    type: 'filesystem'
+  },
+  entry: {
+    app: [
+      JS_FILE,
+      CSS_FILE
+    ]
+  },
+  output: {
+    filename: '[name].js',
+    path: path.resolve(DIST_DIR)
+  },
+  module: {
+    rules: [{
+      test: /\.jsx?$/,
+      // exclude: /node_modules/,
+      use: {
+        loader: 'babel-loader',
+        options: {
+          presets: [
+            '@babel/preset-env',
+            '@babel/react',
+            {
               plugins: [
-                [
-                  '@babel/transform-react-jsx'
-                ]
-              ],
-              cacheDirectory: true,
-              cacheCompression: true
+                '@babel/plugin-proposal-class-properties'
+              ]
             }
+          ], // Preset used for env setup
+          plugins: [
+            ['@babel/transform-react-jsx']
+          ],
+          cacheDirectory: true,
+          cacheCompression: true
+        }
+      }
+    },
+    {
+      test: /\.s?css$/,
+      use: [{
+        loader: MiniCssExtractPlugin.loader
+      },
+      {
+        loader: 'css-loader',
+        options: {
+          sourceMap: false
+        }
+      },
+      {
+        loader: 'resolve-url-loader'
+      },
+      {
+        loader: 'sass-loader',
+        options: {
+          sourceMap: false
+        }
+      },
+      {
+        loader: 'postcss-loader',
+        options: {
+          postcssOptions: {
+            plugins: ['autoprefixer']
           }
-        },
-        // css
-        {
-          test: /\.s?css$/,
-          use: [
-            {
-              loader: MiniCssExtractPlugin.loader
-            },
-            {
-              loader: 'css-loader',
-              options: {
-                sourceMap: false
-              }
+        }
+      }
+      ]
+    },
+    {
+      test: /\.(png|webp|jpg|jpeg|gif|svg)$/,
+      use: [{
+        loader: 'img-optimize-loader',
+        options: {
+          name: '[name].[ext]',
+          outputPath: IMG_DIR_CONFIG,
+          compress: {
+            // This will take more time and get smaller images.
+            mode: 'low', // 'lossless', 'high', 'low'
+            disableOnDevelopment: true,
+            // convert to webp
+            webp: false,
```

Comment:
If you will set this option to true all images will be converted to webp, but some browsers doesn't support it so let's keep it false